Makefile

```diff
@@ -123,7 +123,7 @@ docker-otelcontribcol:
 
 .PHONY: otelcontribcol
 otelcontribcol:
-	GO111MODULE=on CGO_ENABLED=0 go build -o ./bin/otelcontribcol_$(GOOS)_$(GOARCH) $(BUILD_INFO) ./cmd/otelcontribcol
+	GO111MODULE=on CGO_ENABLED=1 go build -o ./bin/otelcontribcol_$(GOOS)_$(GOARCH) $(BUILD_INFO) ./cmd/otelcontribcol
```

Comment:
Considering how specialized cuda is, and that the collector by default needs to be built for many different architectures, it seems like we wouldn't want to package it by default in the binary and should have a custom build for it? In that case, this could probably stay as `CGO_ENABLED=0`, which I'm presuming is so the built binary can run on as many different platforms (e.g., alpine) as possible.

Reply:
I totally understand the demand to avoid CGO and I would like to avoid the use of CUDA that requires CGO if possible. However, it's inevitable to use NVML (CUDA) for GPU monitoring and because [it bundles all at once](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/cmd/otelcontribcol/components.go#L74-L87), this PR requires to turn on CGO.

Otherwise, it should have special flag to include this receiver in the Makefile to enable CGO for the build of this receiver.

Reply:
I think a flag like that makes sense, it seems like building cuda by default will be hard to maintain.

Reply:
Not to mention it'd force every contributor to be able to compile this receiver. We could provide build docker images that make it simpler but still would be downgrade from the current dev exp. Perhaps it can be added to the repo but not enabled by default. May be we can also add another command entry point at `cmd/otelcontribcol-cuda` for a special build and people who need it could build from that. If there is enough demand, we could publish special docker images for cuda.

Reply:
@owais I created another Make target and go build flag for this receiver in next couple of commits to avoid forcing all contributors to build this receiver. However, as written in my last issue comment, the default docker image used in CircleCI and staticcheck doesn't let the build pass because of the lack of dependent libraries and the bug in staticcheck. At least, the former will be solved with the special docker image that you mention, but I don't come up with any good solution to handle staticcheck issue.